plugins/org.jboss.tools.playground.easymport.maven/src/org/jboss/tools/playground/easymport/maven/MavenProjectConfigurator.java

```diff
@@ -32,16 +34,61 @@
 import org.eclipse.jface.wizard.IWizard;
 import org.eclipse.m2e.core.MavenPlugin;
 import org.eclipse.m2e.core.internal.IMavenConstants;
+import org.eclipse.m2e.core.internal.lifecyclemapping.discovery.LifecycleMappingDiscoveryRequest;
 import org.eclipse.m2e.core.internal.project.ProjectConfigurationManager;
 import org.eclipse.m2e.core.project.IProjectConfigurationManager;
 import org.eclipse.m2e.core.project.LocalProjectScanner;
 import org.eclipse.m2e.core.project.MavenProjectInfo;
 import org.eclipse.m2e.core.project.MavenUpdateRequest;
 import org.eclipse.m2e.core.project.ResolverConfiguration;
+import org.eclipse.m2e.core.ui.internal.wizards.LifecycleMappingDiscoveryHelper;
+import org.eclipse.m2e.core.ui.internal.wizards.MappingDiscoveryJob;
+import org.eclipse.ui.internal.wizards.datatransfer.EasymportJob;
 import org.eclipse.ui.wizards.datatransfer.ProjectConfigurator;
 
 public class MavenProjectConfigurator implements ProjectConfigurator {
 	
+	private static class CumulativeMappingDiscoveryJob extends MappingDiscoveryJob {
+		private static CumulativeMappingDiscoveryJob INSTANCE;
+		private Set<IProject> toProcess;
```

Comment:
Thanks for noticing it. I'll fix it as part of next iteration.
More than the small memory leak (which is actually size of a pointer * max number of projects in workspace), I believe this can have an important impact on the performance of the connector discovery operations.